apps/total-typescript/src/trpc/routers/bookmarks.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  addBookmark,
+  BookmarkInputSchema,
+  ReturnedBookmarkSchema,
+  BookmarkForBook,
+  ReturnedBookmarkForBookSchema,
+  deleteBookmark,
+  getBookmarksForUser,
+  getBookmarkByResourceId,
+  getLastestBookmarkedResources,
+} from '@/lib/bookmarks'
+import {TRPCError} from '@trpc/server'
+import {publicProcedure, router} from '@skillrecordings/skill-lesson'
+import {getToken} from 'next-auth/jwt'
+import {z} from 'zod'
+
+export const bookmarksRouter = router({
+  getBookmark: publicProcedure
+    .input(z.object({id: z.string()}))
+    .query(async ({ctx, input}) => {
+      const token = await getToken({req: ctx.req})
+      if (!token) {
+        throw new TRPCError({
+          code: 'UNAUTHORIZED',
+          message: 'User not logged in',
+        })
+      }
+
+      const bookmark = await getBookmarkByResourceId({
+        userId: token.id as string,
+        resourceId: input.id,
+      })
+
+      return ReturnedBookmarkSchema.nullable().parse(bookmark)
```

Comment:
parsing bookmarks strips `Prisma.JsonValue` types out and gives us objects that we expect on the front end. This works well but not sure how one generally deals with the `Prisma.JsonValue` type otherwise.